Wrappers/Python/test/test_dataexample.py

```diff
@@ -158,70 +159,95 @@ class TestRemoteData(unittest.TestCase):
+        mock_response.read.return_value = zipped_bytes
         mock_response.__enter__.return_value = mock_response
         mock_urlopen.return_value = mock_response
 
     @patch('cil.utilities.dataexample.urlopen')
     def test_unzip_remote_data(self, mock_urlopen):
-        self.mock_urlopen(mock_urlopen)
-        dataexample.REMOTEDATA._download_and_extract_from_url('.')
-        self.assertTrue(os.path.isfile(self.tmp_file))
+        
+        shapes_path = os.path.join(dataexample.CILDATA.data_dir, dataexample.TestData.SHAPES)
+
+        with NamedTemporaryFile(suffix = '.zip') as tf:
+            tmp_path = os.path.dirname(tf.name)
+            tmp_dir = os.path.splitext(os.path.basename(tf.name))[0]
+            with ZipFile(tf.name, mode='w') as zip_file:
+                zip_file.write(shapes_path, arcname=dataexample.TestData.SHAPES)
+                
+            with open(tf.name, 'rb') as zip_file:
+                zipped_bytes = zip_file.read()
+        
+        self.mock_urlopen(mock_urlopen, zipped_bytes)
+        dataexample.REMOTEDATA._download_and_extract_from_url(os.path.join(tmp_path, tmp_dir))
+
+        self.assertTrue(os.path.isfile(os.path.join(tmp_path, tmp_dir, dataexample.TestData.SHAPES)))
+
+        if os.path.exists(os.path.join(tmp_path,tmp_dir)):
+            shutil.rmtree(os.path.join(tmp_path,tmp_dir)) 
+        
 
     @patch('cil.utilities.dataexample.input', return_value='n')    
     @patch('cil.utilities.dataexample.urlopen')
     def test_download_data_input_n(self, mock_urlopen, input):
-        self.mock_urlopen(mock_urlopen)
+        
+        shapes_path = os.path.join(dataexample.CILDATA.data_dir, dataexample.TestData.SHAPES)
+
+        with NamedTemporaryFile(suffix = '.zip') as tf:
+            tmp_path = os.path.dirname(tf.name)
+            tmp_dir = os.path.splitext(os.path.basename(tf.name))[0]
+            with ZipFile(tf.name, mode='w') as zip_file:
+                zip_file.write(shapes_path, arcname=dataexample.TestData.SHAPES)
+                
+            with open(tf.name, 'rb') as zip_file:
+                    zipped_bytes = zip_file.read()
 
-        data_list = ['WALNUT','USB','KORN','SANDSTONE']
-        for data in data_list:
-             # redirect print output
+        self.mock_urlopen(mock_urlopen, zipped_bytes)
+
+        for data in self.data_list:
+            # redirect print output
             capturedOutput = StringIO()                 
-            sys.stdout = capturedOutput  
+            sys.stdout = capturedOutput 
             test_func = getattr(dataexample, data)
-            test_func.download_data('.')
-            
-            self.assertFalse(os.path.isfile(self.tmp_file))
+            test_func.download_data(os.path.join(tmp_path, tmp_dir))
+            self.assertFalse(os.path.isfile(os.path.join(tmp_path, tmp_dir, test_func.FOLDER, dataexample.TestData.SHAPES)))
             self.assertEqual(capturedOutput.getvalue(),'Download cancelled\n')
```

Comment:
These asserts should include a message saying where it failed. So in this case which dataset in the loop it got to.